pkg/controller/seed-controller-manager/mla/resources.go

```diff
@@ -397,3 +451,84 @@ func getOrgByProjectID(ctx context.Context, client ctrlruntimeclient.Client, gra
 	}
 	return grafanaClient.GetOrgById(ctx, uint(id))
 }
+
+// GatewayCACreator returns a function to create the ECDSA-based CA to be used for MLA Gateway.
+func GatewayCACreator() reconciling.NamedSecretCreatorGetter {
+	return func() (string, reconciling.SecretCreator) {
+		return resources.MLAGatewayCASecretName, func(se *corev1.Secret) (*corev1.Secret, error) {
+			if se.Data == nil {
+				se.Data = map[string][]byte{}
+			}
+
+			if data, exists := se.Data[resources.MLAGatewayCACertKey]; exists {
+				certs, err := certutil.ParseCertsPEM(data)
+				if err != nil {
+					return nil, fmt.Errorf("failed to parse certificate %s from existing secret %s: %v",
+						resources.MLAGatewayCACertKey, resources.MLAGatewayCASecretName, err)
+				}
+				if !resources.CertWillExpireSoon(certs[0]) {
+					return se, nil
+				}
+			}
+
+			cert, key, err := certificates.GetECDSACACertAndKey()
+			if err != nil {
+				return nil, fmt.Errorf("failed to generate MLA CA: %v", err)
+			}
+			se.Data[resources.MLAGatewayCACertKey] = cert
+			se.Data[resources.MLAGatewayCAKeyKey] = key
+
+			return se, nil
+		}
+	}
+}
+
+type tlsServerCertCreatorData interface {
```

Comment:
seems that this interface is only there to have an abstraction between the `TemplateData` type and the certificate creator. Since the `TemplateData` has the underlying method, and this interface is only use in here I would remove it and use a `func` type instead. See the comment below.  And where the method `GatewayCertificateCreator` is called, just call the (*TemplateData). GatewayCertificateCreator() directly.